staging/src/k8s.io/apimachinery/pkg/api/resource/amount_test.go

```diff
@@ -88,6 +88,54 @@ func TestInt64AmountAdd(t *testing.T) {
 		}
 	}
 }
+
+func TestInt64AmountMul(t *testing.T) {
+	for _, test := range []struct {
+		a, b, c int64Amount
+		ok      bool
+	}{
+		{int64Amount{value: 100, scale: 1}, int64Amount{value: 10, scale: 2}, int64Amount{value: 10000, scale: 1}, true},
+		{int64Amount{value: 100, scale: 1}, int64Amount{value: 1, scale: 2}, int64Amount{value: 1000, scale: 1}, true},
+		{int64Amount{value: 100, scale: 1}, int64Amount{value: 1, scale: 100}, int64Amount{value: 1, scale: 100}, false},
+		{int64Amount{value: -5, scale: 2}, int64Amount{value: 50, scale: 1}, int64Amount{value: -2500, scale: 1}, true},
+		{int64Amount{value: -5, scale: 2}, int64Amount{value: 5, scale: 2}, int64Amount{value: -25, scale: 2}, true},
+
+		{int64Amount{value: mostPositive, scale: -1}, int64Amount{value: 1, scale: -1}, int64Amount{value: 9223372036854775807, scale: -1}, true},
+		{int64Amount{value: mostPositive, scale: -1}, int64Amount{value: 0, scale: -1}, int64Amount{value: 0, scale: -1}, true},
+		{int64Amount{value: mostPositive / 10, scale: 1}, int64Amount{value: 10, scale: 0}, int64Amount{value: mostPositive, scale: -1}, false},
```

Comment:
These last three test cases are very careful to test the cases where the algorithm returns correct answers. Instead, they need to test the cases where the algorithm overflows. I would expect that some overflow can be prevented by changing scale.

Reply:
> I would expect that some overflow can be prevented by changing scale.

In other operators for an `int64Amount`, such as `Add` and `Sub`, they don't change a scale if overflow or underflow would result, and then they would return `false`.

So, I implemented `Mul` so that `Mul` returns `false`, not changing a scale.

Note: In Quantity, Mul avoids overflow by changing a scale.